Source/VisualStudioTools/Private/VSTestAdapterCommandlet.cpp

```diff
@@ -2,134 +2,153 @@
 
 #include "VSTestAdapterCommandlet.h"
 
-#include <Runtime/Core/Public/Async/TaskGraphInterfaces.h>
-#include <Runtime/Core/Public/Containers/Ticker.h>
-#include <fstream>
+#include "Runtime/Core/Public/Async/TaskGraphInterfaces.h"
+#include "Runtime/Core/Public/Containers/Ticker.h"
 #include <string>
 
 #include "VisualStudioTools.h"
 
-int32 UVSTestAdapterCommandlet::Main(const FString& Params)
+static constexpr auto ListTestsParam = TEXT("listtests");
+static constexpr auto RunTestsParam = TEXT("runtests");
+static constexpr auto TestResultsFileParam = TEXT("testresultfile");
+static constexpr auto HelpParam = TEXT("help");
+
+static void GetAllTests(TArray< FAutomationTestInfo >& OutTestList)
 {
-	TArray<FString> Tokens;
-	TArray<FString> Switches;
-	TMap<FString, FString> ParamVals;
+	FAutomationTestFramework& Framework = FAutomationTestFramework::GetInstance();
+	Framework.GetValidTestNames(OutTestList);
+}
 
-	// Functionality for Unreal Engine Test Adapter.
-	ParseCommandLine(*Params, Tokens, Switches, ParamVals);
-	FAutomationTestFramework::GetInstance().SetRequestedTestFilter( EAutomationTestFlags::PriorityMask | EAutomationTestFlags::ProductFilter | EAutomationTestFlags::SmokeFilter | EAutomationTestFlags::PerfFilter | EAutomationTestFlags::StressFilter | EAutomationTestFlags::NegativeFilter );	
-	if ( ParamVals.Contains( TEXT( "listtests" ) ) )
+static void ReadTestsFromFile(const FString& InFile, TArray< FAutomationTestInfo >& OutTestList)
```

Comment:
Still some formatting issues in a few places. 
Not blocking, we can fix them later, so only do it if you have more changes planned already.

Reply:
Will address in future PR soon.